src/parser/monk/brewmaster/modules/spells/azeritetraits/StraightNoChaser.js

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import { Line as LineChart } from 'react-chartjs-2';
+import Analyzer from 'parser/core/Analyzer';
+import SPELLS from 'common/SPELLS';
+import { calculateAzeriteEffects } from 'common/stats';
+import StatTracker from 'parser/shared/modules/StatTracker';
+import TraitStatisticBox from 'interface/others/TraitStatisticBox';
+import SpellLink from 'common/SpellLink';
+import AbilityTracker from 'parser/shared/modules/AbilityTracker';
+import { formatNumber, formatPercentage } from 'common/format';
+
+import SpellUsable from '../../core/SpellUsable';
+
+const SNC_PROC_CHANCE = 0.08;
+
+export default class StraightNoChaser extends Analyzer {
+  static dependencies = {
+    abilityTracker: AbilityTracker,
+    stats: StatTracker,
+    spellUsable: SpellUsable,
+  };
+
+  armor = 0;
+
+  get isbCasts() {
+    return this.abilityTracker.getAbility(SPELLS.IRONSKIN_BREW.id).casts;
+  }
+
+  get resets() {
+    return this.spellUsable._SNCResets;
+  }
+
+  get avgArmor() {
+    return this.armor * this.selectedCombatant.getBuffUptime(SPELLS.STRAIGHT_NO_CHASER_BUFF.id) / this.owner.fightDuration;
+  }
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTrait(SPELLS.STRAIGHT_NO_CHASER.id);
+    if(!this.active) {
+      return;
+    }
+
+    this.armor = this.selectedCombatant.traitsBySpellId[SPELLS.STRAIGHT_NO_CHASER.id].reduce((sum, rank) => {
+      const [armor] = calculateAzeriteEffects(SPELLS.STRAIGHT_NO_CHASER.id, rank);
+      return sum + armor;
+    }, 0);
+
+    this.stats.add(SPELLS.STRAIGHT_NO_CHASER.id, {
+      armor: this.armor,
+    });
+  }
+
+  // currently literal copypasta from the `MasteryValue` module. Once
+  // we're ready to start moving to next this will get replaced with a
+  // generic implementation.
+  plot() {
+    // not the most efficient, but close enough and pretty safe
+    function binom(n, k) {
+      if(k > n) {
+        return null;
+      }
+      if(k === 0) {
+        return 1;
+      }
+
+      return n / k * binom(n-1, k-1);
+    }
+
+    // pmf of the binomial distribution
+    const reset_prob = (i) => binom(this.isbCasts, i) * Math.pow(SNC_PROC_CHANCE, i) * Math.pow(1 - SNC_PROC_CHANCE, this.isbCasts - i);
+
+    const reset_probs = Array.from({length: this.isbCasts}, (_x, i) => {
```

Comment:
I guess I understand the point here, but wouldn't the graph look a little nicer if it was from like.. first half only? It seems to reach the 0 asymptote very quickly and looks a little harder to read this way? Just an idea, if it's not viable or doesn't make sense, keep it this way.